pkg/tarmak/provider/amazon/amazon.go

```diff
@@ -600,6 +621,30 @@ func (a *Amazon) VolumeType(typeIn string) (typeOut string, err error) {
 	if typeIn == clusterv1alpha1.VolumeTypeSSD {
 		return "gp2", nil
 	}
-	// TODO: Validate custom instance type here
+
+	found := false
+	for _, t := range a.awsVolumeTypes() {
+		if t == typeIn {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		return "", fmt.Errorf("'%s' is not a supported volume type", typeIn)
+	}
+
 	return typeIn, nil
 }
+
+func (a *Amazon) awsVolumeTypes() []string {
+	return []string{"io1", "gp2", "st1", "sc1"}
+}
+
+func (a *Amazon) awsInstanceTypes() []string {
```

Comment:
I think you are missing some newer instance types, is there a way to make this dynamic and check this through an AWS call?

Reply:
I've had a look and I can't find a straight forward way of achieving this. I've updated this list though